### DISCUSSION TITLE
Unused parameters

### DISCUSSION THREAD

**Discussion by davidnmbond** [simple-odata-client/Simple.OData.Client#847] [General]
In the spirit of "no errors, no warnings, no messages and all unit tests should pass"...

The following parameters are currently unused and are public:
- Simple.OData.Client.TypeCacheResolver.GetAnyField(string fieldName, **bool includeNonPublic = false**)
- Simple.OData.Client.TypeExtensions.GetAnyField(this Type type, string fieldName, **bool includeNonPublic = false**)

I am unaware of the history here.  It is possible that they are "old DNA" and are provided for backward compatibility reasons?  I propose adding NEW method signatures without those unused parameters and to mark the old methods [Obsolete], with a message to the user explaining why.  Any comments welcome prior to me starting the feature branch?

**Out of discussion scope**: There are also a number of unused private/protected parameters, which can be removed as a lower priority, later.

**Comment by object** [comment]
The library's aim is to provide simple access to OData functionality. I don't think it should expose anything else. I no longer remember why these methods were introduce but either they were used earlier or they were added for the reason of completeness (not a sufficienlty good reason). So feel free to remove them.
